### PR TITLE
Vision cloud label detection

### PR DESCRIPTION
**Pull request by DennisAlund**
This is an implementation of the [cloud based API in ML Kit for Firebase](https://firebase.google.com/docs/ml-kit/label-images#example-cloud-labels)

I found that the cloud based APIs were missing and wrote one straight off due to personal need in a planned workshop. I haven't reported this in any feature request, but I figured that it would be part of the roadmap.

### CODE REVIEW

**Comment by DennisAlund** [comment]
The issues are noted.

**Comment by DennisAlund** [comment]
Hey, @bparrishMines I see that you've been writing most of the ml-vision plugin. I tried to add support for cloud labeling of images. But I'm a bit stumped by the failing build here. 

It builds and runs fine for me locally and I can't make out what action I could take to fix the breaking build.

Any ideas or pointers?

**Comment by bparrishMines** [comment]
Hi @DennisAlund,

Could you do a `git pull`? That may help with some of the errors for the PLUGIN_SHARDING check.

Also, this error may be the culprit for the build issue:

```
/var/folders/xb/469gd9j941bc7v3mltn7zgb40000gn/T/cirrus-ci-build/packages/firebase_ml_vision/ios/Classes/CloudLabelDetector.m:43:37: error: no visible @interface for 'NSNumber' declares the selector 'uintValue'
      detector.maxResults = [maxResults uintValue];
```

**Comment by DennisAlund** [comment]
Thanks for spotting that @bparrishMines, I coded the ios part a bit in the dark on my pc.

**Comment by DennisAlund** [comment]
Any other ideas on the current build errors @bparrishMines?

**Comment by bparrishMines** [comment]
Yea, if you click on `details` for either of the failed `PLUGIN_SHARDING` checks, you can see where the errors are coming from. You can see that there are 6 errors if you scroll to the bottom. There are also a bunch of warnings, but I don't believe they are caused by your code, so they can be ignored. Lmk if you don't see them.

**Comment by bparrishMines** [comment]
Hi @DennisAlund, are you still working on this PR?

**Comment by DennisAlund** [comment]
Hey, I didn't have time to look into it yet. Would need to try it out with xcode since it is the objectivec code that is causing issues. But I haven't had a chance to do that. Would appreciate input from the community if there's someone who's got xcode and a bit of time.

File: packages/firebase_ml_vision/lib/src/vision_cloud_detector_options.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+class VisionCloudDetectorOptions {
+  /// Constructor for [VisionCloudDetectorOptions].
+  ///
+  const VisionCloudDetectorOptions(
+      {this.maxResults = 10, this.modelType = MODEL_TYPE_STABLE})
+      : assert(maxResults > 0),
+        assert(
+            modelType == MODEL_TYPE_LATEST || modelType == MODEL_TYPE_STABLE);
+
+  static const int MODEL_TYPE_STABLE = 1;
```
Comment (bparrishMines) [inline packages/firebase_ml_vision/lib/src/vision_cloud_detector_options.dart]: In dart we typically prefer `lowerCamelCase` for constants.

File: packages/firebase_ml_vision/lib/src/vision_cloud_detector_options.dart [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  static const int MODEL_TYPE_LATEST = 2;
+
+  /// The number of results to be returned.
+  /// Defaults to 10.
```
Comment (bparrishMines) [inline packages/firebase_ml_vision/lib/src/vision_cloud_detector_options.dart]: A space is needed under top comment summary.

File: packages/firebase_ml_vision/lib/src/vision_cloud_detector_options.dart (hunk 1 above)
Comment (DennisAlund) [reply packages/firebase_ml_vision/lib/src/vision_cloud_detector_options.dart]: Fixed in 4969a6dff

File: packages/firebase_ml_vision/lib/src/vision_cloud_detector_options.dart (hunk 2 above)
Comment (DennisAlund) [reply packages/firebase_ml_vision/lib/src/vision_cloud_detector_options.dart]: Fixed in 4969a6dff

**Comment by bparrishMines** [comment]
Thanks! I have a branch that will bump the version and this will get published with it.

**Pull request merged**